lib/Cake/Model/Model.php

```diff
@@ -3438,6 +3456,38 @@ public function query($sql) {
 	}
 
 /**
+ * Builds and execute the query. Return DataSource object to retrieve the result explicitly using `fetchResult()`
+ *
+ * @param array $query Option fields (conditions / fields / joins / limit / offset / order / page / group / callbacks)
+ * @param int $recursive Number of levels of association
+ * @return mixed DataSource object if query executes with no problem, null on failure
+ */
+	public function runQuery($query = array(), $recursive = null) {
+		$db = clone $this->getDataSource();
+		if (!$db) {
+			return null;
+		}
+
+		$queryData = $this->buildQuery('all', $query);
+		if (is_null($queryData)) {
```

Comment:
@cithukyaw It's more about code consistency rather than speed. Throughout the codebase we use `=== null`.